internal/appsec/emitter/sharedsec/actions.go

```diff
@@ -63,15 +74,94 @@ type (
 	//    Such constructors would receive the full appsec config and rules, so that they would be able to build
 	//    specific blocking actions.
 	GRPCWrapper func(map[string][]string) (uint32, error)
+
+	// blockActionParams are the dynamic parameters to be provided to a "block_request"
+	// action type upon invocation
+	blockActionParams struct {
+		GRPCStatusCode *int   `mapstructure:"grpc_status_code,omitempty"`
+		StatusCode     int    `mapstructure:"status_code"`
+		Type           string `mapstructure:"type,omitempty"`
+	}
+	// redirectActionParams are the dynamic parameters to be provided to a "redirect_request"
+	// action type upon invocation
+	redirectActionParams struct {
+		Location   string `mapstructure:"location,omitempty"`
+		StatusCode int    `mapstructure:"status_code"`
+	}
 )
 
-// Blocking returns true if the action object represents a request blocking action
-func (a *Action) Blocking() bool {
-	return a.blocking
+func (a *HTTPAction) Blocking() bool              { return true }
+func (a *HTTPAction) EmitData(op dyngo.Operation) { dyngo.EmitData(op, a) }
+
+func (a *GRPCAction) Blocking() bool              { return true }
+func (a *GRPCAction) EmitData(op dyngo.Operation) { dyngo.EmitData(op, a) }
+
+func (a *StackTraceAction) Blocking() bool              { return false }
+func (a *StackTraceAction) EmitData(op dyngo.Operation) { dyngo.EmitData(op, a) }
+
+// NewStackTraceAction creates an action for the "stacktrace" action type
+func NewStackTraceAction(params map[string]any) Action {
+	id, ok := params["stack_id"]
+	if !ok {
+		log.Debug("appsec: could not read stack_id parameter for stack_trace action")
+		return nil
+	}
+	event := stacktrace.NewEvent(stacktrace.ExploitEvent, "", "")
+	if event == nil {
+		log.Debug("appsec: could not generate stacktrace event")
+		return nil
+	}
+	event.ID = id.(string)
```

Comment:
So you skipped the mapstructure part? Can you do a try cast instead of only a cast?